packages/loopring_v3/test/testUserStakingPool.ts

```diff
@@ -1,266 +1,307 @@
-        await exchangeTestUtil.advanceBlockTimestamp(90 * 24 * 60 * 60);
-      });
-      it("set protocolfee as 120", async () => {
-        const amount = new BN(120);
-        await exchangeTestUtil.transferBalance(
-          protocolfee.address,
-          "LRC",
-          amount
-        );
-      });
-      it("should get user A claimed equal as expected", async () => {
-        const userclaimed = await userstaking.claim({ from: owner1 });
-        const eventArr: any = await exchangeTestUtil.getEventsFromContract(
-          userstaking,
-          "LRCRewarded",
-          web3.eth.blockNumber
-        );
-        const items = eventArr.map((eventObj: any) => {
-          const reward = eventObj.args.amount;
-          // 120 * 70%  * (180 / (180 + 120 + 90)) == 35 as staking reward
-          // as A's staking time is 30 + 30 + 90 days, B's staking time is 30 + 90 days and C's staking time is 90 days
-
-          // 34 for round down when div
-          assert(
-            reward == 35 || reward == 34,
-            "User claimed should equal to expected"
-          );
-        });
-      });
-      it("should get user B claimed equal as expected", async () => {
-        const userclaimed = await userstaking.claim({ from: owner2 });
-        const eventArr: any = await exchangeTestUtil.getEventsFromContract(
-          userstaking,
-          "LRCRewarded",
-          web3.eth.blockNumber
-        );
-        const items = eventArr.map((eventObj: any) => {
-          const reward = eventObj.args.amount;
-          // 120 * 70%  * (180 / (180 + 120 + 90)) == 28 as staking reward
-          assert(
-            reward == 27 || reward == 28 || reward == 29,
-            "User claimed should equal to expected"
-          );
-        });
-      });
-      it("should get user C claimed equal as expected", async () => {
-        const userclaimed = await userstaking.claim({ from: owner3 });
-        const eventArr: any = await exchangeTestUtil.getEventsFromContract(
-          userstaking,
-          "LRCRewarded",
-          web3.eth.blockNumber
-        );
-        const items = eventArr.map((eventObj: any) => {
-          const reward = eventObj.args.amount;
-          // 120 * 70%  * (180 / (180 + 120 + 90)) == 21 as staking reward
-          assert(
-            reward == 20 || reward == 21 || reward == 22,
-            "User claimed should equal to expected"
-          );
-        });
-      });
-      it("User A withdraw all", async () => {
-        await userstaking.withdraw(0, { from: owner1 });
-      });
-      it("User B withdraw all", async () => {
-        await userstaking.withdraw(0, { from: owner2 });
-      });
-      it("User C withdraw all", async () => {
-        await userstaking.withdraw(0, { from: owner3 });
+  describe("when no protocol fee vault is set, user Charles", () => {
+    const amount = new BN("1000" + "000000000000000000", 10);
```

Comment:
Not a fan of `new BN(web3.utils.toWei("1000", "ether"))`?

Reply:
Nice to know this. I'll refactor it in my next PR which will introduce more test cases.